CHANGELOG.md

```diff
@@ -1,8 +1,40 @@
 # Change Log
 
+## Unreleased
+- `CompactBlockProcessor` now processes compact blocks from the lightwalletd server with **Spend-before-Sync** algorithm 
+  (i.e. non-linear order). This feature shortens the time after which a wallet's spendable balance can be used.
+- The block synchronization mechanism is about one-third faster thanks to the optimized 
+`CompactBlockProcessor.SYNC_BATCH_SIZE`. Issue **#1206**.
+
+### Removed
+- `CompactBlockProcessor.ProcessorInfo.lastSyncHeight` which the SDK is no longer able to provide because of the new
+  **SpendBeforeSync** synchronization algorithm adoption. Use `CompactBlockProcessor.ProcessorInfo.overallSyncRange` 
+  which contains all blocks in case of `SpendBeforeSync` synchronization algorithm. No internal change was made in 
+  case of older Linear synchronization algorithm.
+- `CompactBlockProcessor.ProcessorInfo.isSyncing`. Use `Synchronizer.status` instead.
+- `CompactBlockProcessor.ProcessorInfo.syncProgress`. Use `Synchronizer.progress` instead.
+- `alsoClearBlockCache` parameter from rewind functions of `Synchronizer` and `CompactBlockProcessor` as it take no 
+  effect on the current rewind functionality result.
+- Internally, we removed access to the shared block table from the Kotlin layer, which resulted in eliminating these 
+  APIs:
+  - `SdkSynchornizer.findBlockHash()`
+  - `SdkSynchornizer.findBlockHashAsHex()`
+
+### Changed
+- `CompactBlockProcessor.quickRewind()` and `CompactBlockProcessor.rewindToNearestHeight()` now might fail due to
+  internal changes in getting scanned height. Thus, these functions return `Boolean` results.
+- `Synchronizer.new()` requires a new `walletInitMode` parameter of type `WalletInitMode`, which describes wallet
+  initialization mode. See related function and sealed class documentation.
+
+### Fixed
+- `Synchronizer.getMemos()` now correctly returns a flow of strings for sent and received transactions. Issue **#1154**.
+- `CompactBlockProcessor` now triggers transaction polling while block synchronization is in progress as expected.
+  Clients will be notified briefly after every new transaction is discovered via `Synchronizer.transactions` API.
```

Comment:
```suggestion
  Clients will be notified shortly after every new transaction is discovered via `Synchronizer.transactions` API.
```

Reply:
Thank you, Daira, for proofreading our texts!